feat(ui): core components created/updated for startv2 page auto creation from core | Issue #63

DebkantaMondal opened the pull request:
> Assigning to upstream maintainers for review. #63 [part1 - startv2 website]

### Overview

- Made lower to higher level components in Core 
- using Core, we can build the actual startV2  or any website with full control through props and services, models using in general core components

### Output

1. Dark Mode:

**Desktop View**
<img width="1459" height="875" alt="Screenshot 2025-11-13 at 1 19 02 PM" src="https://github.com/user-attachments/assets/28fe8116-7868-4713-aec4-77bfda3ef17e" />
**Mobile View**
<img width="250" height="350" alt="Screenshot 2025-11-13 at 1 20 32 PM" src="https://github.com/user-attachments/assets/aa1c371a-146a-4a87-8cdc-0a59a8d12434" />
<img width="250" height="350" alt="Screenshot 2025-11-13 at 1 20 47 PM" src="https://github.com/user-attachments/assets/57f6496c-30d3-4b4e-9382-67f32b807ca7" />

2. Light Mode:

**Desktop View**
<img width="1460" height="867" alt="Screenshot 2025-11-13 at 1 23 04 PM" src="https://github.com/user-attachments/assets/58e261f2-9ea3-4e6e-8729-7d09820c0007" />
**Mobile View**
<img width="250" height="350" alt="Screenshot 2025-11-13 at 1 23 16 PM" src="https://github.com/user-attachments/assets/e58c815d-30f0-49e5-9534-f6fc85f49be5" />
<img width="250" height="350" alt="Screenshot 2025-11-13 at 1 23 25 PM" src="https://github.com/user-attachments/assets/f61d1f9b-c910-4315-bc57-d6ff3fe24543" />


### Lint Result

<img width="792" height="254" alt="Screenshot 2025-11-13 at 1 25 20 PM" src="https://github.com/user-attachments/assets/0236aef8-8f02-4595-883f-637a52330330" />

### Technical Usage 

Mentioned in startV2 component in docs - `localhost:4200/startV2 by running pnpm "serve docs"`

### Test Result 

Environment | Result
-- | --
macOS + pnpm "serve docs" | Works smoothly
Linux | Works smoothly
Dark / Light themes | Colors consistent



Signed-off-by: Debkanta Mondal ([debkantaroni01@gmail.com](mailto:debkantaroni01@gmail.com))

DebkantaMondal committed:
feat(ui): core components created/updated for startv2 page auto creation from core

dr460nf1r3 commented:
Hey! First impression while not having looked at it locally is that the search bar color (white) is quite striking and does not fit the color theme imo. Is it a design choice?

I would also prefer sticking to the new signal functions instead of the decorators (input, output, viewchild) to ensure we're able to run in reactive contexts. Angular schematics can do this automatically, there's no need to do it by hand (though you might need a few adjustments where things are accessed).

DebkantaMondal committed:
feat(ui): core components created/updated for startv2 page auto creation from core | Issue #63

DebkantaMondal commented:
> Hey! First impression while not having looked at it locally is that the search bar color (white) is quite striking and does not fit the color theme imo. Is it a design choice?
> 
> I would also prefer sticking to the new signal functions instead of the decorators (input, output, viewchild) to ensure we're able to run in reactive contexts. Angular schematics can do this automatically, there's no need to do it by hand (though you might need a few adjustments where things are accessed).

Now everywhere input,output,viewchild signal functions is used and updated instead of decorators. 

Yes, Search component is a design choice, if anyone want to customize design can do from their component. But as you have suggested, i have changed its default color pattern for both light and dark theme which is quite similar to existing search component.

### Output:

**Light Theme:**
<img width="350" height="250" alt="Screenshot 2025-11-15 at 9 41 03 AM" src="https://github.com/user-attachments/assets/3957a4bb-adc8-492f-9e25-66741e01e2f1" />

On Focus:
<img width="350" height="250" alt="Screenshot 2025-11-15 at 9 41 10 AM" src="https://github.com/user-attachments/assets/59ba7e28-2bfd-460d-9161-a8db9fb0078c" />

**Dark Theme:**
<img width="350" height="250" alt="Screenshot 2025-11-15 at 9 40 30 AM" src="https://github.com/user-attachments/assets/c8618116-202d-49c4-a6e0-80ddd15b44fb" />

On Focus:
<img width="350" height="250" alt="Screenshot 2025-11-15 at 9 40 43 AM" src="https://github.com/user-attachments/assets/93ca0830-e3b2-4dcb-930e-b0e3cfd16453" />

### Lint Result:
<img width="350" height="250" alt="Screenshot 2025-11-13 at 1 25 20 PM" src="https://github.com/user-attachments/assets/c3b071a8-3c80-44cc-88eb-1314778801aa" />


Signed-off-by: Debkanta Mondal ([debkantaroni01@gmail.com](mailto:debkantaroni01@gmail.com))

DebkantaMondal committed:
feat(ui): core components created/updated for startv2 page auto creation from core | Issue #63 | lint fixed

DebkantaMondal committed:
feat(ui): updated eslint and lint fixed ci

DebkantaMondal committed:
feat(ui): updated eslint and lint fixed ci

DebkantaMondal committed:
feat(ui): reverted eslint

DebkantaMondal committed:
feat(ui): revert

DebkantaMondal committed:
feat(ui): reverted

DebkantaMondal committed:
feat(ui): pnpm-yaml back

dr460nf1r3 commented:
Is it passing the lint locally?

DebkantaMondal commented:
> Is it passing the lint locally?

Yes. pnpm lint is passing all cases.
<img width="673" height="299" alt="Screenshot 2025-11-15 at 2 56 30 PM" src="https://github.com/user-attachments/assets/57f18cb8-6251-44e7-a872-6a72fdc25acf" />

dr460nf1r3 commented:
Ah you even sent the screenshot in the first post, my bad. I'll try to look into this, though I need to find some time first.

DebkantaMondal committed:
feat(ui): ci-test

DebkantaMondal committed:
feat(ui): ci-lint-test

DebkantaMondal committed:
feat(ui): ci-test

DebkantaMondal commented:
i have changed few:  pre-commit directly in ci and lint passed now. could u please check ci file changes and if all ok approve for deployment

DebkantaMondal commented:
is this pr need more time to approve? @dr460nf1r3 @Vo1dSh4d0w

dr460nf1r3 commented:
I'm sorry, real life tasks got me busy. I wanted to give a good review. I've now reviewed it locally. My findings:
- clicking into the search bar moves the search bar down a bit — that's visual distraction I would like to prevent if there is no reason to do so
- If quick to do, please make the start V2 integrated with the page transitions the other tabs have when switching
- When switching between tabs, there is a really weird effect happening when transitioning, here a screenshot:
	<img width="1014" height="203" alt="image" src="https://github.com/user-attachments/assets/0d304212-ff27-47c1-9d71-295d93052e21" />
  Basically, it looks like the content is entirely gone, and blur is applied where it should not be. This is not something we have on the current dev version (https://dev.garuda-ng.pages.dev/)
- Please attach the footer to the bottom of the page
	<img width="1386" height="325" alt="image" src="https://github.com/user-attachments/assets/0ea57cce-4df7-422c-bfed-feb4b93b4e11" />
- When clicking the open card button, the card jumps. This seems to be due to initially opening the more information dropdown, hiding it by default and opening on click only should probably get rid of this. 

Once these are resolved, I'm happy to merge this. Thanks for taking the time, and sorry for taking so long again.

DebkantaMondal commented:
Sure. I will check and update the PR soon in upcoming weekend.

DebkantaMondal committed:
Merge branch 'dev' into feature/debkanta-reusable-core

DebkantaMondal committed:
feat(ui): updated

DebkantaMondal committed:
fix(ui): updated cases of footer

DebkantaMondal commented:
Fixed all the points of Footer and Search Box Issues as mentioned. @dr460nf1r3 @Vo1dSh4d0w . Please check and Merge.

DebkantaMondal commented:
@dr460nf1r3 could u please check and merge the pr? Mentioned Issues are already fixed. @Vo1dSh4d0w

dr460nf1r3 commented:
Thanks for fixing it, I need to find a short moment to review it.

dr460nf1r3 committed:
feat(docs): animationstate + jumping fix for startv2

dr460nf1r3 commented:
Thanks for the contribution! :)
